docs/rules/arazzo/parameters-unique.md

```diff
@@ -0,0 +1,93 @@
+---
+slug: /docs/cli/rules/arazzo/parameters-unique
+---
+
+# parameters-unique
+
+Requires the `parameters` list not to include duplicate parameters.
+
+| ARAZZO | Compatibility |
+| ------ | ------------- |
+| 1.0.0  | ✅            |
+
+```mermaid
+flowchart TD
+
+Root ==> workflows --> workflow --> parameters
+
+style Contact fill:#codaf9,stroke:#0044d4,stroke-width:5px
+```
+
+```mermaid
+flowchart TD
+
+Root ==> x-parameters
+
+style Contact fill:#codaf9,stroke:#0044d4,stroke-width:5px
+```
+
+```mermaid
+flowchart TD
+
+Root ==> workflows --> workflow --> steps --> step --> parameters
+
+style Contact fill:#codaf9,stroke:#0044d4,stroke-width:5px
+```
+
+## API design principles
+
+A list of `parameters` that are applicable to step or all steps described under workflow should not be duplicated to avoid shadow parameter override.
+
+## Configuration
+
+| Option   | Type   | Description                                             |
+| -------- | ------ | ------------------------------------------------------- |
+| severity | string | Possible values: `off`, `warn`, `error`. Default `off`. |
+
+An example configuration:
+
+```yaml
+arazzoRules:
+  parameters-unique: error
+```
+
+## Examples
+
+Given the following configuration:
+
+```yaml
+arazzoRules:
+  parameters-unique: error
+```
+
+Example of an **incorrect** license:
+
+```yaml Object example
+workflows:
+  - workflowId: get-museum-hours
+    parameters:
+      - in: header
+        name: Authorization
+        value: Main Og==
+      - in: header
+        name: Authorization
+        value: Basic Og==
+```
+
+Example of a **correct** license:
+
+```yaml Object example
+workflows:
+  - workflowId: get-museum-hours
+    parameters:
+      - in: header
+        name: Authorization
+        value: Basic Og==
+      - in: header
+        name: Auth
+        value: Main Og==
+```
```

Comment:
I'm trying to see the difference between these and I see one has a header "Auth" and the other one has two "Authorization" headers. What does this have to do with "licenses"?

Reply:
I wanted to illustrate that parameters can't have the same name under one `parameters` section, maybe the correct example can be different.

Reply:
The schema should handle these particular constraints without additional rules.

Reply:
@jeremyfiel could you explain how exactly the schema enforces unique parameter names?

Reply:
hmm.. i suppose you may have a point here.. while it has `uniqueItems: true`, it doesn't mean you can't have one parameter with two different values. 

```yaml
parameters:
  - name: test
    in: header
    value: foo
  - name: test
    in: header
    value: bar
```

You're right!

this was proposed in JSON Schema as a new vocabulary but hasn't been included in the latest draft.  It's currently a custom keyword implementation.  https://github.com/json-schema-org/json-schema-vocabularies/issues/22

Reply:
Then checking the uniqueness with the rules still makes sense.

Reply:
I adjusted the example to be more "obvious" in what it illustrates.